### PR TITLE
Fakenodo Service

### PR DESCRIPTION
**Pull request by clameaitu**
Falta:
- Fix las funciones del servicio (todo menos create deposition)
- Tests de unidad
- Tests de locust
- Test de selenium (crear un dataset)
- Se ha creado una incidencia #141

### CODE REVIEW

**Comment by davidgonmar** [comment]
## Test Coverage Report
  **conftest.py**: 100%

### **auth**
  **forms.py**: 100%
  **models.py**: 88.6800%
  **repositories.py**: 84.3100%
  **routes.py**: 32.6700%
  **services.py**: 71.1100%

### **chatbot**
  **forms.py**: 0%
  **models.py**: 0%
  **repositories.py**: 0%
  **routes.py**: 48.00%
  **services.py**: 46.6700%

### **dataset**
  **api.py**: 100%
  **forms.py**: 78.00%
  **models.py**: 81.100%
  **repositories.py**: 78.5700%
  **routes.py**: 31.6500%
  **services.py**: 54.6100%

### **explore**
  **forms.py**: 100%
  **repositories.py**: 12.2400%
  **routes.py**: 40.0%
  **services.py**: 71.4300%

### **fakenodo**
  **forms.py**: 0%
  **models.py**: 70.0%
  **repositories.py**: 53.8500%
  **routes.py**: 35.2300%
  **services.py**: 42.1900%

### **featuremodel**
  **forms.py**: 0%
  **models.py**: 86.00%
  **repositories.py**: 67.8600%
  **routes.py**: 80.0%
  **services.py**: 82.1400%

### **flamapy**
  **forms.py**: 0%
  **models.py**: 100%
  **repositories.py**: 0%
  **routes.py**: 31.9400%
  **services.py**: 0%

### **hubfile**
  **forms.py**: 0%
  **models.py**: 73.9100%
  **repositories.py**: 92.3100%
  **routes.py**: 17.0700%
  **services.py**: 72.4100%

### **notepad**
  **forms.py**: 100%
  **models.py**: 88.8900%
  **repositories.py**: 100%
  **routes.py**: 84.2100%
  **services.py**: 100%

### **profile**
  **forms.py**: 100%
  **models.py**: 76.9200%
  **repositories.py**: 100%
  **routes.py**: 76.6700%
  **services.py**: 100%

### **public**
  **routes.py**: 100%

### **team**
  **routes.py**: 80.0%

### **webhook**
  **forms.py**: 0%
  **models.py**: 0%
  **repositories.py**: 0%
  **routes.py**: 0%
  **services.py**: 0%

### **zenodo**
  **forms.py**: 0%
  **models.py**: 100%
  **repositories.py**: 100%
  **routes.py**: 70.0%
  **services.py**: 35.8500%

**Comment by clameaitu** [comment]
#111

**Comment by davidgonmar** [comment]
## Test Coverage Report
  **conftest.py**: 100%

### **auth**
  **forms.py**: 100%
  **models.py**: 88.6800%
  **repositories.py**: 84.3100%
  **routes.py**: 32.6700%
  **services.py**: 71.1100%

### **chatbot**
  **forms.py**: 0%
  **models.py**: 0%
  **repositories.py**: 0%
  **routes.py**: 48.00%
  **services.py**: 46.6700%

### **dataset**
  **api.py**: 100%
  **forms.py**: 78.00%
  **models.py**: 81.100%
  **repositories.py**: 78.5700%
  **routes.py**: 31.6500%
  **services.py**: 54.6100%

### **explore**
  **forms.py**: 100%
  **repositories.py**: 12.2400%
  **routes.py**: 40.0%
  **services.py**: 71.4300%

### **fakenodo**
  **forms.py**: 0%
  **models.py**: 70.0%
  **repositories.py**: 53.8500%
  **routes.py**: 35.2300%
  **services.py**: 42.1900%

### **featuremodel**
  **forms.py**: 0%
  **models.py**: 86.00%
  **repositories.py**: 67.8600%
  **routes.py**: 80.0%
  **services.py**: 82.1400%

### **flamapy**
  **forms.py**: 0%
  **models.py**: 100%
  **repositories.py**: 0%
  **routes.py**: 31.9400%
  **services.py**: 0%

### **hubfile**
  **forms.py**: 0%
  **models.py**: 73.9100%
  **repositories.py**: 92.3100%
  **routes.py**: 17.0700%
  **services.py**: 72.4100%

### **notepad**
  **forms.py**: 100%
  **models.py**: 88.8900%
  **repositories.py**: 100%
  **routes.py**: 84.2100%
  **services.py**: 100%

### **profile**
  **forms.py**: 100%
  **models.py**: 76.9200%
  **repositories.py**: 100%
  **routes.py**: 76.6700%
  **services.py**: 100%

### **public**
  **routes.py**: 100%

### **team**
  **routes.py**: 80.0%

### **webhook**
  **forms.py**: 0%
  **models.py**: 0%
  **repositories.py**: 0%
  **routes.py**: 0%
  **services.py**: 0%

### **zenodo**
  **forms.py**: 0%
  **models.py**: 100%
  **repositories.py**: 100%
  **routes.py**: 70.0%
  **services.py**: 35.8500%

**Comment by davidgonmar** [comment]
## Test Coverage Report
  **conftest.py**: 100%

### **auth**
  **forms.py**: 100%
  **models.py**: 88.6800%
  **repositories.py**: 84.3100%
  **routes.py**: 32.6700%
  **services.py**: 71.1100%

### **chatbot**
  **forms.py**: 0%
  **models.py**: 0%
  **repositories.py**: 0%
  **routes.py**: 48.00%
  **services.py**: 46.6700%

### **dataset**
  **api.py**: 100%
  **forms.py**: 78.00%
  **models.py**: 81.100%
  **repositories.py**: 78.5700%
  **routes.py**: 31.6500%
  **services.py**: 54.6100%

### **explore**
  **forms.py**: 100%
  **repositories.py**: 12.2400%
  **routes.py**: 40.0%
  **services.py**: 71.4300%

### **fakenodo**
  **forms.py**: 0%
  **models.py**: 70.0%
  **repositories.py**: 53.8500%
  **routes.py**: 35.2300%
  **services.py**: 42.1900%

### **featuremodel**
  **forms.py**: 0%
  **models.py**: 86.00%
  **repositories.py**: 67.8600%
  **routes.py**: 80.0%
  **services.py**: 82.1400%

### **flamapy**
  **forms.py**: 0%
  **models.py**: 100%
  **repositories.py**: 0%
  **routes.py**: 31.9400%
  **services.py**: 0%

### **hubfile**
  **forms.py**: 0%
  **models.py**: 73.9100%
  **repositories.py**: 92.3100%
  **routes.py**: 17.0700%
  **services.py**: 72.4100%

### **notepad**
  **forms.py**: 100%
  **models.py**: 88.8900%
  **repositories.py**: 100%
  **routes.py**: 84.2100%
  **services.py**: 100%

### **profile**
  **forms.py**: 100%
  **models.py**: 76.9200%
  **repositories.py**: 100%
  **routes.py**: 76.6700%
  **services.py**: 100%

### **public**
  **routes.py**: 100%

### **team**
  **routes.py**: 80.0%

### **webhook**
  **forms.py**: 0%
  **models.py**: 0%
  **repositories.py**: 0%
  **routes.py**: 0%
  **services.py**: 0%

### **zenodo**
  **forms.py**: 0%
  **models.py**: 100%
  **repositories.py**: 100%
  **routes.py**: 70.0%
  **services.py**: 35.8500%

**Comment by davidgonmar** [comment]
## Test Coverage Report
  **conftest.py**: 100%

### **auth**
  **forms.py**: 100%
  **models.py**: 88.6800%
  **repositories.py**: 84.3100%
  **routes.py**: 32.6700%
  **services.py**: 71.1100%

### **chatbot**
  **forms.py**: 0%
  **models.py**: 0%
  **repositories.py**: 0%
  **routes.py**: 48.00%
  **services.py**: 46.6700%

### **dataset**
  **api.py**: 100%
  **forms.py**: 78.00%
  **models.py**: 81.100%
  **repositories.py**: 78.5700%
  **routes.py**: 31.6500%
  **services.py**: 54.6100%

### **explore**
  **forms.py**: 100%
  **repositories.py**: 12.2400%
  **routes.py**: 40.0%
  **services.py**: 71.4300%

### **fakenodo**
  **forms.py**: 0%
  **models.py**: 70.0%
  **repositories.py**: 53.8500%
  **routes.py**: 35.2300%
  **services.py**: 42.1900%

### **featuremodel**
  **forms.py**: 0%
  **models.py**: 86.00%
  **repositories.py**: 67.8600%
  **routes.py**: 80.0%
  **services.py**: 82.1400%

### **flamapy**
  **forms.py**: 0%
  **models.py**: 100%
  **repositories.py**: 0%
  **routes.py**: 31.9400%
  **services.py**: 0%

### **hubfile**
  **forms.py**: 0%
  **models.py**: 73.9100%
  **repositories.py**: 92.3100%
  **routes.py**: 17.0700%
  **services.py**: 72.4100%

### **notepad**
  **forms.py**: 100%
  **models.py**: 88.8900%
  **repositories.py**: 100%
  **routes.py**: 84.2100%
  **services.py**: 100%

### **profile**
  **forms.py**: 100%
  **models.py**: 76.9200%
  **repositories.py**: 100%
  **routes.py**: 76.6700%
  **services.py**: 100%

### **public**
  **routes.py**: 100%

### **team**
  **routes.py**: 80.0%

### **webhook**
  **forms.py**: 0%
  **models.py**: 0%
  **repositories.py**: 0%
  **routes.py**: 0%
  **services.py**: 0%

### **zenodo**
  **forms.py**: 0%
  **models.py**: 100%
  **repositories.py**: 100%
  **routes.py**: 70.0%
  **services.py**: 35.8500%

**Comment by davidgonmar** [comment]
## Test Coverage Report
  **conftest.py**: 100%

### **auth**
  **forms.py**: 100%
  **models.py**: 88.6800%
  **repositories.py**: 84.3100%
  **routes.py**: 32.6700%
  **services.py**: 71.1100%

### **chatbot**
  **forms.py**: 0%
  **models.py**: 0%
  **repositories.py**: 0%
  **routes.py**: 48.00%
  **services.py**: 46.6700%

### **dataset**
  **api.py**: 100%
  **forms.py**: 78.00%
  **models.py**: 81.100%
  **repositories.py**: 78.5700%
  **routes.py**: 31.6500%
  **services.py**: 54.6100%

### **explore**
  **forms.py**: 100%
  **repositories.py**: 12.2400%
  **routes.py**: 40.0%
  **services.py**: 71.4300%

### **fakenodo**
  **forms.py**: 0%
  **models.py**: 70.0%
  **repositories.py**: 53.8500%
  **routes.py**: 35.2300%
  **services.py**: 42.1900%

### **featuremodel**
  **forms.py**: 0%
  **models.py**: 86.00%
  **repositories.py**: 67.8600%
  **routes.py**: 80.0%
  **services.py**: 82.1400%

### **flamapy**
  **forms.py**: 0%
  **models.py**: 100%
  **repositories.py**: 0%
  **routes.py**: 31.9400%
  **services.py**: 0%

### **hubfile**
  **forms.py**: 0%
  **models.py**: 73.9100%
  **repositories.py**: 92.3100%
  **routes.py**: 17.0700%
  **services.py**: 72.4100%

### **notepad**
  **forms.py**: 100%
  **models.py**: 88.8900%
  **repositories.py**: 100%
  **routes.py**: 84.2100%
  **services.py**: 100%

### **profile**
  **forms.py**: 100%
  **models.py**: 76.9200%
  **repositories.py**: 100%
  **routes.py**: 76.6700%
  **services.py**: 100%

### **public**
  **routes.py**: 100%

### **team**
  **routes.py**: 80.0%

### **webhook**
  **forms.py**: 0%
  **models.py**: 0%
  **repositories.py**: 0%
  **routes.py**: 0%
  **services.py**: 0%

### **zenodo**
  **forms.py**: 0%
  **models.py**: 100%
  **repositories.py**: 100%
  **routes.py**: 70.0%
  **services.py**: 35.8500%

**Comment by davidgonmar** [comment]
## Test Coverage Report
  **conftest.py**: 100%

### **auth**
  **forms.py**: 100%
  **models.py**: 88.6800%
  **repositories.py**: 84.3100%
  **routes.py**: 32.6700%
  **services.py**: 71.1100%

### **chatbot**
  **forms.py**: 0%
  **models.py**: 0%
  **repositories.py**: 0%
  **routes.py**: 48.00%
  **services.py**: 46.6700%

### **dataset**
  **api.py**: 100%
  **forms.py**: 78.00%
  **models.py**: 81.8900%
  **repositories.py**: 78.5700%
  **routes.py**: 31.6500%
  **services.py**: 54.6100%

### **explore**
  **forms.py**: 100%
  **repositories.py**: 12.2400%
  **routes.py**: 40.0%
  **services.py**: 71.4300%

### **fakenodo**
  **forms.py**: 0%
  **models.py**: 70.0%
  **repositories.py**: 53.8500%
  **routes.py**: 35.2300%
  **services.py**: 42.1900%

### **featuremodel**
  **forms.py**: 0%
  **models.py**: 86.00%
  **repositories.py**: 67.8600%
  **routes.py**: 80.0%
  **services.py**: 82.1400%

### **flamapy**
  **forms.py**: 0%
  **models.py**: 100%
  **repositories.py**: 0%
  **routes.py**: 31.9400%
  **services.py**: 0%

### **hubfile**
  **forms.py**: 0%
  **models.py**: 73.9100%
  **repositories.py**: 92.3100%
  **routes.py**: 17.0700%
  **services.py**: 72.4100%

### **notepad**
  **forms.py**: 100%
  **models.py**: 88.8900%
  **repositories.py**: 100%
  **routes.py**: 84.2100%
  **services.py**: 100%

### **profile**
  **forms.py**: 100%
  **models.py**: 76.9200%
  **repositories.py**: 100%
  **routes.py**: 76.6700%
  **services.py**: 100%

### **public**
  **routes.py**: 100%

### **team**
  **routes.py**: 80.0%

### **webhook**
  **forms.py**: 0%
  **models.py**: 0%
  **repositories.py**: 0%
  **routes.py**: 0%
  **services.py**: 0%

### **zenodo**
  **forms.py**: 0%
  **models.py**: 100%
  **repositories.py**: 100%
  **routes.py**: 70.0%
  **services.py**: 35.8500%

File: migrations/versions/b9a2ee27aa79_.py
Comment (rafcasceb) [inline migrations/versions/b9a2ee27aa79_.py:1]: Estoy no habría que cambiarlo, creo. Lo arreglo yo como mantenimiento global de la aplicación, fuera de este issue en concreto.

**Review by rafcasceb — CHANGES_REQUESTED** [review]

**Comment by davidgonmar** [comment]
## Test Coverage Report
  **conftest.py**: 100%

### **auth**
  **forms.py**: 100%
  **models.py**: 88.6800%
  **repositories.py**: 84.3100%
  **routes.py**: 32.6700%
  **services.py**: 71.1100%

### **chatbot**
  **forms.py**: 0%
  **models.py**: 0%
  **repositories.py**: 0%
  **routes.py**: 48.00%
  **services.py**: 46.6700%

### **dataset**
  **api.py**: 100%
  **forms.py**: 78.00%
  **models.py**: 78.5200%
  **repositories.py**: 80.0%
  **routes.py**: 32.100%
  **services.py**: 55.2600%

### **explore**
  **forms.py**: 100%
  **repositories.py**: 12.2400%
  **routes.py**: 44.4400%
  **services.py**: 71.4300%

### **fakenodo**
  **forms.py**: 0%
  **models.py**: 70.0%
  **repositories.py**: 53.8500%
  **routes.py**: 35.2300%
  **services.py**: 42.1900%

### **featuremodel**
  **forms.py**: 0%
  **models.py**: 86.00%
  **repositories.py**: 67.8600%
  **routes.py**: 80.0%
  **services.py**: 82.1400%

### **flamapy**
  **forms.py**: 0%
  **models.py**: 100%
  **repositories.py**: 0%
  **routes.py**: 31.9400%
  **services.py**: 0%

### **hubfile**
  **forms.py**: 0%
  **models.py**: 73.9100%
  **repositories.py**: 92.3100%
  **routes.py**: 17.0700%
  **services.py**: 72.4100%

### **notepad**
  **forms.py**: 100%
  **models.py**: 88.8900%
  **repositories.py**: 100%
  **routes.py**: 84.2100%
  **services.py**: 100%

### **profile**
  **forms.py**: 100%
  **models.py**: 76.9200%
  **repositories.py**: 100%
  **routes.py**: 76.6700%
  **services.py**: 100%

### **public**
  **routes.py**: 95.4500%

### **team**
  **routes.py**: 80.0%

### **webhook**
  **forms.py**: 0%
  **models.py**: 0%
  **repositories.py**: 0%
  **routes.py**: 0%
  **services.py**: 0%

### **zenodo**
  **forms.py**: 0%
  **models.py**: 100%
  **repositories.py**: 100%
  **routes.py**: 70.0%
  **services.py**: 35.8500%

File: migrations/versions/b9a2ee27aa79_.py
Comment (rafcasceb) [reply migrations/versions/b9a2ee27aa79_.py:1]: Listo.

**Review by rafcasceb — APPROVED** [review]
Ya está todo perfecto.

**Pull request merged**